cmake/cuda.cmake

```diff
@@ -21,13 +21,44 @@ cas_variable_cuda_architectures(GINKGO_CUDA_ARCH_FLAGS
     ARCHITECTURES ${GINKGO_CUDA_ARCHITECTURES}
     UNSUPPORTED "20" "21")
 
+if (${CMAKE_CXX_COMPILER_ID} MATCHES "PGI|NVHPC")
+    find_package(NVHPC REQUIRED
```

Comment:
Does this need to be required in this case, or we silently disable CUDA if not found?

Reply:
not sure about PGI.
For NVHPC, I will assume it is always shipped with the cuda toolkit

Reply:
the problem is the `nvc++` compiler is identified as `PGI` at least on Perlmutter...